### DISCUSSION TITLE
Optional AbiParameter name

### DISCUSSION THREAD

**Discussion by unholypanda** [wagmi-dev/abitype#88] [Q&A] [answered]
I noticed in #52 that an intentional decision was made to make the `AbiParameter['name']` optional. I'm wondering if this is the correct type representation for this field.

As far as I can tell, the `name` key is always present on the object for input and output parameters, the value is always of type string, and unnamed parameters are denoted with an empty string.

I didn't see any mention of this field being optional in the Solidity ABI specification, but I'll admit that I only tried looking this up in the last few versions of the Solidity docs.

**Comment by unholypanda** [comment]
Also, I wasn't sure if this should be categorized as `Q&A` or `Ideas` so I went with `Q&A` assuming I'm wrong or am missing context on the motivation for this optional type.

**Comment by tmm** [comment, marked as answer]
Good points! We saw some folks using ABIs with wagmi, where they removed `name: ''` to make their code less verbose (or it wasn't defined in the first place - not sure where they got those ABIs from). Rather than force folks to add `name: ''` everywhere, we made it optional as `name: ''` seems nearly the same as `name?: undefined`.

**Reply by unholypanda** [reply]
Thanks! That makes sense and I appreciate the added context for what motivated the change.
